ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -42,6 +52,29 @@ export default class PermissionConnect extends Component {
     originName: this.props.originName,
   }
 
+  componentDidUpdate (prevProps) {
+    const { permissionsRequest, domains, permissionsRequestId } = this.props
+    const { originName, page } = this.state
+
+    if (!permissionsRequestId && prevProps.permissionsRequestId && page !== null) {
+      const permissionDataForDomain = domains && domains[originName] || {}
+      const permissionsForDomain = permissionDataForDomain.permissions || []
+      const prevPermissionDataForDomain = prevProps.domains && prevProps.domains[originName] || {}
+      const prevPermissionsForDomain = prevPermissionDataForDomain.permissions || []
+      const addedAPermission = permissionsForDomain.length > prevPermissionsForDomain.length
+      if (addedAPermission) {
+        this.redirectFlow(true)
+      } else {
+        this.redirectFlow(false)
+      }
+    } else if (permissionsRequestId && prevProps.permissionsRequestId && permissionsRequestId !== prevProps.permissionsRequestId) {
```

Comment:
I think there should be a `&& page !== null` here as well, or the "redirecting to..." page doesn't work correctly (it swaps back to page 1 of the next request). If there are multiple queued requests, that is.

Reply:
Or alternatively, start with a `if (page === null) { return }` to save adding this to each condition.